### PR TITLE
[doc] Add specifications for calendriers.txt

### PR DESCRIPTION
**Pull request by woshilapin**
I copy/pasting an example here of a `calendar.txt` and a `calendar_dates.txt`, as well as the expected results in `calendriers.xml`.

### `calendar.txt`

```csv
service_id,monday,tuesday,wednesday,thursday,friday,saturday,sunday,start_date,end_date
service_one,0,0,0,0,0,1,1,20190710,20190720
```

### `calendar_dates.txt`

```csv
service_id,date,exception_type
service_one,20190714,1
service_one,20190715,0
service_two,20191225,0
```

### `calendriers.xml`
```xml
<?xml version="1.0" encoding="UTF-8"?>                          
<PublicationDelivery
    version="1.09:FR-NETEX_LIGNE-2.1-1.0"
    xmlns="http://www.netex.org.uk/netex"
    xmlns:core="http://www.govtalk.gov.uk/core"
    xmlns:gml="http://www.opengis.net/gml/3.2"
    xmlns:ifopt="http://www.ifopt.org.uk/ifopt"
    xmlns:siri="http://www.siri.org.uk/siri"
    xmlns:xlink="http://www.w3.org/1999/xlink"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://www.netex.org.uk/netex">                        
  <PublicationTimestamp>2019-04-03T17:19:00</PublicationTimestamp>
  <ParticipantRef>Participant</ParticipantRef>
  <dataObjects>                                                                                                                              
    <GeneralFrame id="FR:GeneralFrame:NETEX_CALENDRIER:LOC" version="any">   
      <ValidBetween>                                                                                                             
        <FromDate>2019-07-09T00:00:00Z</FromDate>               
        <ToDate>2019-08-07T23:59:59Z</ToDate>                                                                              
      </ValidBetween>           
      <members>                                                                                                                  
        <!-- Service from the 10 to 20 of July 2019 only on weekend days -->                 
        <!-- The 14th which is Sunday is removed -->                         
        <!-- The 15th which is a Monday is added -->
        <DayType id="FR:DayType:service_one:LOC" version="any">                                
          <properties>           
            <PropertyOfDay>                                                                                    
              <DaysOfWeek>Saturday</DaysOfWeek>
            </PropertyOfDay>
            <PropertyOfDay>                                                                                    
              <DaysOfWeek>Sunday</DaysOfWeek>
            </PropertyOfDay>                                                                                   
          </properties>                  
        </DayType>          
        <!-- Only to define the 25th of December 2019 -->                                      
        <DayType id="FR:DayType:service_two:LOC" version="any" />
        <OperatingPeriod id="FR:OperatingPeriod:service_one:LOC" version="any">
          <FromDate>2019-07-10T00:00:00Z</FromDate>       
          <ToDate>2019-07-20T23:59:59Z</ToDate>
        </OperatingPeriod>
        <DayTypeAssignment id="FR:DayTypeAssignment:service_one:LOC" version="any" order="0">
          <OperatingPeriodRef ref="FR:OperatingPeriod:service_one:LOC" version="any" />
          <DayTypeRef ref="FR:DayType:service_one:LOC" version="any" />
        </DayTypeAssignment>
        <DayTypeAssignment id="FR:DayTypeAssignment:service_one_1:LOC" version="any" order="1">
          <Date>2019-07-14</Date>
          <DayTypeRef ref="FR:DayType:service_one:LOC" version="any" />
          <isAvailable>false</isAvailable>
        </DayTypeAssignment>
        <DayTypeAssignment id="FR:DayTypeAssignment:service_one_2:LOC" version="any" order="2">
          <Date>2019-07-15</Date>
          <DayTypeRef ref="FR:DayType:service_one:LOC" version="any" />
          <isAvailable>true</isAvailable>
        </DayTypeAssignment>
        <DayTypeAssignment id="FR:DayTypeAssignment:service_two_1:LOC" version="any" order="1">
          <Date>2019-12-25</Date>
          <DayTypeRef ref="FR:DayType:service_two:LOC" version="any" />
          <!-- isAvailable is omitted, true is default -->
        </DayTypeAssignment>
      </members>
    </GeneralFrame>
  </dataObjects>
</PublicationDelivery>
```

### CODE REVIEW

File: src/documentation/ntfs_to_netex_france_specs.md [hunk 1]
```diff
@@ -283,3 +284,90 @@ Line/@id | lines.txt | line_id | see [id formatting](#id-of-objects)
...
+Netex field | NTFS file | NTFS field | Note
+--- | --- | --- | ---
+DayTypeAssignement/@id | calendar.txt | service_id | see [id formatting](#id-of-objects)
+DayTypeAssignement/@version | | | fixed value `any`
+DayTypeAssignement/@order | | | fixed value `0`
+DayTypeAssignement/OperatingPeriodRef/@ref | calendar.txt | service_id | see [id formatting](#id-of-objects) with `OperatingPeriod` as object type
+DayTypeAssignement/DayTypeRef/@ref | calendar.txt | service_id | see [id formatting](#id-of-objects) with `DayType` as object type
+
+### DayTypeAssignement for `calendar_dates.txt` exceptions
+
```
Comment (prhod) [inline src/documentation/ntfs_to_netex_france_specs.md]: Need to add a sort order for exceptions. What about:
`calendar_dates are sorted by date prior to processing to keep a consistent order.`

File: src/documentation/ntfs_to_netex_france_specs.md [hunk 2]
```diff
@@ -283,3 +284,90 @@ Line/@id | lines.txt | line_id | see [id formatting](#id-of-objects)
...
+One `PropertyOfDay` for each weekday of `calendar.txt`. For `service_id` which
+exists only in `calendar_dates.txt`, no `PropertyOfDay` is created (in NeTEx
+France profile, no `PropertyOfDay` means all the days of the week are active).
+
+`DaysOfWeek` can take one of the following values: `Monday`, `Tuesday`,
+`Wednesday`, `Thursday`, `Friday`, `Saturday`, or `Sunday`.
+
+### OperatingPeriod
+
```
Comment (prhod) [inline src/documentation/ntfs_to_netex_france_specs.md]: `An OperatingPeriod is only created for service_id used in calendars.txt.`

File: src/documentation/ntfs_to_netex_france_specs.md (hunk 1 above)
Comment (woshilapin) [reply src/documentation/ntfs_to_netex_france_specs.md]: Done in 138790f96d3559a33e13eb909d88da9ea499f553.

File: src/documentation/ntfs_to_netex_france_specs.md (hunk 2 above)
Comment (woshilapin) [reply src/documentation/ntfs_to_netex_france_specs.md]: Done in 138790f96d3559a33e13eb909d88da9ea499f553.

**Comment by woshilapin** [comment]
> Can you check about UICOperatingPeriod ?

So, both the following (and equivalent) XML passes the `xmllint` validation.

```xml
<members>                                                                              
  <DayType id="FR:DayType:service_one:LOC" version="any" />                            
  <ServiceCalendar id="FR:ServiceCalendar:service_one:LOC" version="any">              
      <dayTypes>                                                                       
          <DayTypeRef ref="FR:DayType:service_one:LOC" />                              
      </dayTypes>                                                                      
      <operatingPeriods>                                                               
          <UicOperatingPeriod id="FR:UicOperatingPeriod:service_one:LOC" version="any">
              <FromDate>2019-07-10T00:00:00Z</FromDate>                                
              <ValidDayBits>0100101000100101</ValidDayBits>                            
          </UicOperatingPeriod>                                                        
      </operatingPeriods>                                                              
  </ServiceCalendar>                                                                   
</members>                                                                             

```

```xml
<members>                                                                      
  <DayType id="FR:DayType:service_one:LOC" version="any" />                    
  <ServiceCalendar id="FR:ServiceCalendar:service_one:LOC" version="any">      
      <dayTypes>                                                               
          <DayTypeRef ref="FR:DayType:service_one:LOC" />                      
      </dayTypes>                                                              
      <operatingPeriods>                                                       
          <OperatingPeriodRef ref="FR:UicOperatingPeriod:service_one:LOC" />   
      </operatingPeriods>                                                      
  </ServiceCalendar>                                                           
  <UicOperatingPeriod id="FR:UicOperatingPeriod:service_one:LOC" version="any">
      <FromDate>2019-07-10T00:00:00Z</FromDate>                                
      <ValidDayBits>0100101000100101</ValidDayBits>                            
  </UicOperatingPeriod>                                                        
</members>                                                                     
```

Note how in the second I use `OperatingPeriodRef` to refer to a `UicOperatingPeriod` (so it does seems to work in the end?). This made test again a structure based on `DayTypeAssignment` which indeed passes the `xmllint` validation (this XML tag order got me again it seems).

```xml
<members>                                                                              
  <DayType id="FR:DayType:service_one:LOC" version="any" />                            
  <ServiceCalendar id="FR:ServiceCalendar:service_one:LOC" version="any">              
      <dayTypes>                                                                       
          <DayTypeRef ref="FR:DayType:service_one:LOC" />                              
      </dayTypes>                                                                      
      <operatingPeriods>                                                               
          <UicOperatingPeriod id="FR:UicOperatingPeriod:service_one:LOC" version="any">
              <FromDate>2019-07-10T00:00:00Z</FromDate>                                
              <ValidDayBits>0100101000100101</ValidDayBits>                            
          </UicOperatingPeriod>                                                        
      </operatingPeriods>                                                              
  </ServiceCalendar>                                                                   
</members>                                                                             
```

If I understand correctly, the 3 above version are strictly modelizing the exact same thing (thanks NeTEx!). Which one should we adopt? Maybe the latest one is the simplest and also the closest from IDF profile?

**Comment by woshilapin** [comment]
I changed the specification to integrate `UicOperatingPeriod`. In order to reflect the changes, below is the same example than in the PR first comment... but with the new format now.

### `calendar.txt`

```csv
service_id,monday,tuesday,wednesday,thursday,friday,saturday,sunday,start_date,end_date
service_one,0,0,0,0,0,1,1,20190710,20190720
```

### `calendar_dates.txt`

```csv
service_id,date,exception_type
service_one,20190714,1
service_one,20190715,0
service_two,20191225,0
```

### `calendriers.xml`
```xml
<?xml version="1.0" encoding="UTF-8"?>                          
<PublicationDelivery
    version="1.09:FR-NETEX_LIGNE-2.1-1.0"
    xmlns="http://www.netex.org.uk/netex"
    xmlns:core="http://www.govtalk.gov.uk/core"
    xmlns:gml="http://www.opengis.net/gml/3.2"
    xmlns:ifopt="http://www.ifopt.org.uk/ifopt"
    xmlns:siri="http://www.siri.org.uk/siri"
    xmlns:xlink="http://www.w3.org/1999/xlink"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://www.netex.org.uk/netex">                        
  <PublicationTimestamp>2019-04-03T17:19:00</PublicationTimestamp>
  <ParticipantRef>Participant</ParticipantRef>
  <dataObjects>                                                                                                                              
    <GeneralFrame id="FR:GeneralFrame:NETEX_CALENDRIER:LOC" version="any">   
      <ValidBetween>                                                                                                             
        <FromDate>2019-07-09T00:00:00Z</FromDate>               
        <ToDate>2019-08-07T23:59:59Z</ToDate>                                                                              
      </ValidBetween>           
      <members>                                                                                                                  
        <!-- Service from the 10 to 20 of July 2019 only on weekend days -->                 
        <!-- The 14th which is Sunday is removed -->                         
        <!-- The 15th which is a Monday is added -->
        <DayType id="FR:DayType:service_one:LOC" version="any" />
        <!-- Only to define the 25th of December 2019 -->                                      
        <DayType id="FR:DayType:service_two:LOC" version="any" />
        <DayTypeAssignment id="FR:DayTypeAssignment:service_one:LOC" version="any" order="0">
          <OperatingPeriodRef ref="FR:OperatingPeriod:service_one:LOC" version="any" />
          <DayTypeRef ref="FR:DayType:service_one:LOC" version="any" />
        </DayTypeAssignment>
        <DayTypeAssignment id="FR:DayTypeAssignment:service_two:LOC" version="any" order="0">
          <OperatingPeriodRef ref="FR:OperatingPeriod:service_two:LOC" version="any" />
          <DayTypeRef ref="FR:DayType:service_two:LOC" version="any" />
        </DayTypeAssignment>
        <UicOperatingPeriod id="FR:OperatingPeriod:service_one:LOC" version="any">
          <FromDate>2019-07-10T00:00:00Z</FromDate>       
          <ValidDayBits>00010100001</ValidDayBits>
        </UicOperatingPeriod>
        <UicOperatingPeriod id="FR:OperatingPeriod:service_two:LOC" version="any">
          <FromDate>2019-12-25T00:00:00Z</FromDate>       
          <ValidDayBits>1</ValidDayBits>
        </UicOperatingPeriod>
      </members>
    </GeneralFrame>
  </dataObjects>
</PublicationDelivery>
```

File: src/documentation/ntfs_to_netex_france_specs.md [hunk 3]
```diff
@@ -283,3 +284,72 @@ Line/@id | lines.txt | line_id | see [id formatting](#id-of-objects)
...
+--- | --- | --- | ---
+UicOperatingPeriod/@id | calendar.txt | service_id | see [id formatting](#id-of-objects) with `OperatingPeriod` as object type
+UicOperatingPeriod/@version | | | fixed value `any`
+UicOperatingPeriod/FromDate | calendar.txt | start_date | use ISO-8601 date-time format, using `T00:00:00Z` for the time
+UicOperatingPeriod/ValidDayBits | | | see (1) below
+
+**(1) Sequence for ValidDayBits**
+
+This is a string sequence of `0` (inactive days) and `1` (active days). The
+sequence should contains as many `0/1` as there is days in the period.
```
Comment (prhod) [inline src/documentation/ntfs_to_netex_france_specs.md]: Sorry to bother you again, but can you make explicit what the first digit stands for ? the start or the end ?

Comment (woshilapin) [reply src/documentation/ntfs_to_netex_france_specs.md]: Done in 315dd5d82a41a33dc1b95b5aec33764d2043984c.

**Comment by woshilapin** [comment]
I just rebased this PR before merging it. It was approved by @prhod already.

**Pull request merged**